### PR TITLE
Added support for creating a recurring contribution for a delayed pay…

### PR DESCRIPTION
**Pull request by Edzelopez**
…ment

### CODE REVIEW

**Comment by Edzelopez** [comment]
The PR is dependent on adding a hook invocation for 
`CRM_Utils_Hook::alterPaymentProcessorParams($this,
      $params,
      $propertyBag
    );`

here: https://github.com/JMAConsulting/ca.civicrm.moneris/blob/vaultwithupdatedrepeat/CRM/Core/Payment/Moneris.php#L92

File: api/v3/Job/SendNotification.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    AND pp.class_name = 'Payment_Moneris'
+    AND DATE(cr.next_sched_contribution_date) = DATE(DATE_ADD(NOW(), INTERVAL $days DAY))";
+  $dao = CRM_Core_DAO::executeQuery($sql)->fetchAll();
+  $cids = [];
+  foreach ($dao as $result) {
+    if (!empty($result['contribution_page_id'])
+      && array_key_exists($result['contribution_page_id'], $contribForms)
+      && !empty($contribForms[$result['contribution_page_id']])) {
+      // This contribution is part of a delayed billing series. Send a notification that payment is due tomorrow.
+      civicrm_api3('Email', 'send', [
```
Comment (seamuslee001) [inline api/v3/Job/SendNotification.php]: do we need to wrap this in a try catch?

File: delayedbilling.php [hunk 2]
```diff
@@ -218,6 +235,32 @@ function delayedbilling_civicrm_pre($op, $objectName, $id, &$params) {
...
+/**
+ * Implements hook_civicrm_post().
+ */
+function delayedbilling_civicrm_post($op, $objectName, $objectId, &$objectRef) {
+  if ($objectName == 'Contribution' && $op == 'edit') {
+    // Check to see if status == Failed.
+    $failedStatus = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Failed');
+    if ($objectRef->contribution_status_id == $failedStatus) {
+      // Check if this was a delayed payment.
+      if (_checkDelayedPayment($objectRef->contribution_page_id)) {
```
Comment (seamuslee001) [inline delayedbilling.php]: Should we also check for the presence of a contribution_recur_id here to avoid the once off failures?

**Comment by monishdeb** [comment]
Tested on cagis.st, working fine.

**Pull request merged**